src/core/types/signalr.ts

```diff
@@ -33,3 +33,128 @@ export type AniDBBanType = {
   http: AniDBBanItemType;
   udp: AniDBBanItemType;
 };
+
+export const enum NetworkAvailability {
+  /**
+   * Shoko was unable to find any network interfaces.
+   */
+  NoInterfaces = 'NoInterfaces',
+
+  /**
+   * Shoko was unable to find any local gateways to use.
+   */
+  NoGateways = 'NoGateways',
+
+  /**
+   * Shoko was able to find a local gateway.
+   */
+  LocalOnly = 'LocalOnly',
+
+  /**
+   * Shoko was able to connect to some internet endpoints in WAN.
+   */
+  PartialInternet = 'PartialInternet',
+
+  /**
+   * Shoko was able to connect to all internet endpoints in WAN.
+   */
+  Internet = 'Internet',
+}
+
+export type AVDumpRestoreType = {
+  type: AVDumpEventTypeEnum.Restore;
+  sessionID: number;
+  videoIDs: number[];
+  commandID: number | null;
+  progress: number;
+  succeededCreqCount: number;
+  failedCreqCount: number;
+  pendingCreqCount: number;
+  ed2ks: number[];
+  startedAt: string;
+};
+
+export type AVDumpEventType = {
+  type: AVDumpEventTypeEnum.Started;
+  sessionID: number;
+  videoIDs: number[];
+  commandID: number | null;
+  progress: number;
+  succeededCreqCount: number;
+  failedCreqCount: number;
+  pendingCreqCount: number;
+  startedAt: string;
+} | {
+  type: AVDumpEventTypeEnum.Progress;
+  sessionID: number;
+  progress: number;
+} | {
+  type: AVDumpEventTypeEnum.CreqUpdate;
+  sessionID: number;
+  succeededCreqCount: number;
+  failedCreqCount: number;
+  pendingCreqCount: number;
+} | {
+  type: AVDumpEventTypeEnum.Message | AVDumpEventTypeEnum.Error | AVDumpEventTypeEnum.ED2KLink;
+  sessionID: number;
+  message: string;
+} | {
+  type:
+    | AVDumpEventTypeEnum.InstalledAVDump
+    | AVDumpEventTypeEnum.InstallingAVDump
+    | AVDumpEventTypeEnum.InvalidCredentials
+    | AVDumpEventTypeEnum.MissingApiKey
+    | AVDumpEventTypeEnum.Timeout;
+} | {
+  type: AVDumpEventTypeEnum.InstallException;
+  message: string;
+  exceptionStackTrace: string;
+} | {
+  type: AVDumpEventTypeEnum.GenericException;
+  sessionID: number;
+  message: string;
+  exceptionStackTrace: string;
+  startedAt: string;
+} | {
+  type: AVDumpEventTypeEnum.Failure;
+  sessionID: number;
+  videoIDs: number[];
+  commandID: number | null;
+  progress: number;
+  ed2ks: number[];
+  message: string;
+  errorMessage?: string;
+  startedAt: string;
+  endedAt: string;
+} | {
+  type: AVDumpEventTypeEnum.Success;
+  sessionID: number;
+  videoIDs: number[];
+  commandID: number | null;
+  progress: number;
+  succeededCreqCount: number;
+  failedCreqCount: number;
+  pendingCreqCount: number;
+  message: string;
+  startedAt: string;
+  endedAt: string;
+};
```

Comment:
How sure are you that this will work without issues?

Reply:
![image](https://github.com/ShokoAnime/Shoko-WebUI/assets/26010946/486dd750-5677-429d-a534-53c73dd7c6ad)

Union operator doesn't work like you think it does

Reply:
Change `seriesId` to `seriesID` and add `as const` at the end of the declaration of `testObj` to get it to qualify as a progress event.

Also, these types are sent from the server, and knowing which fields are available for which events are useful, since we can use the `type` field to narrow down the availablefields using typescript type guards. If i wanted to change it, then i would declare each event separately and export them, then create a union of the now named events.

Reply:
Your example;
![image](https://github.com/ShokoAnime/Shoko-WebUI/assets/7761729/57bfc675-dddc-45b0-92f3-c14ecc4df2e8)
![image](https://github.com/ShokoAnime/Shoko-WebUI/assets/7761729/6c5b10a8-1606-4364-913c-aa0032055c07)

W/ what i just suggested;
![image](https://github.com/ShokoAnime/Shoko-WebUI/assets/7761729/a56e8e3a-3063-46e4-badb-8be4851bff88)
![image](https://github.com/ShokoAnime/Shoko-WebUI/assets/7761729/262aa2d4-4b71-416a-b3be-1893da2e7d2a)